### PR TITLE
add simpler system-probe server listener

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Refactors the system-probe server listener to have a simpler interface and be located in `cmd/system-probe/api/server`.

### Motivation

Split PR from #30936

### Describe how to test/QA your changes

Manually tested locally and in a staging node.

### Additional Notes

Stacked PR on top of #31150

### CODE REVIEW

**Review by guyarb — APPROVED** [review]
LGTM

**Review by GustavoCaso — APPROVED** [review]
LGTM!

**Review by hmahmood — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]
looks good for the only Processes file: `pkg/languagedetection/detector_linux_test.go`

**Review by derekwbrown — APPROVED** [review]
Approved for (2) WKIT files

File: cmd/system-probe/api/server/listener_unix.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	conn, err := net.Listen("unix", socketAddr)
+	if err != nil {
+		return nil, fmt.Errorf("listen: %s", err)
+	}
+
+	if err := os.Chmod(socketAddr, 0720); err != nil {
+		return nil, fmt.Errorf("socket chmod write-only: %s", err)
+	}
```
Comment (guyarb) [inline cmd/system-probe/api/server/listener_unix.go:46]: if write only, why do you set it with executable permissions?

Comment (guyarb) [reply cmd/system-probe/api/server/listener_unix.go:46]: It is not clear to what you're trying to do in this function
Please add documentation for the function

**Review by val06 — APPROVED** [review]

**Pull request merged**